BrainPortal/app/models/boutiques_portal_task.rb

```diff
@@ -72,6 +72,11 @@ def descriptor_for_show_params
     self.boutiques_descriptor
   end
 
+  # Return true if the descriptor have only one mandatory input
+  def have_a_single_mandatory_input_file?(descriptor)
+    descriptor.file_inputs.select{|i| i.optional == false }.size == 1
```

Comment:
Don't check with an explicit comparison `something == false` because that fails for nil. Try it in the console. There are several better ways of doing your check:

```ruby
descriptor.file_inputs.select { |i| ! i.optional }.size == 1
descriptor.file_inputs.reject { |i|   i.optional }.size == 1
```

and the best, using a block for count():

```ruby
descriptor.file_inputs.count { |i| ! i.optional } == 1
```